src/basic/missing_syscall.h

```diff
@@ -406,7 +406,13 @@ static inline int missing_epoll_pwait2(
                 const struct timespec *timeout,
                 const sigset_t *sigset) {
 
-#  if defined(__NR_epoll_pwait2) && HAVE_LINUX_TIME_TYPES_H
+#  if __SIZEOF_LONG__ == 4
+// Someone with an interest in 32bit systems, please have a look at this, and figure out why this hangs on 32bit systems.
+// My educated guess: might be because of issues with the __kernel_timespec translation or because of incorrectly sized sigset_t array.
+#    pragma message "epoll_pwait2() appears to be broken on 32bit archs, someone please have a look!"
+        errno = ENOSYS;
+        return -1;
```

Comment:
Why do you warn all 32bit arch even if `__NR_epoll_pwait2` is not defined? How about the following?
```
#  if defined(__NR_epoll_pwait2) && HAVE_LINUX_TIME_TYPES_H
#    if __SIZEOF_LONG__ == 4
#      pragma message ...
        errno = ENOSYS;
        return -1;
#    else
        (...)
#    endif
#  else
        errno = ENOSYS;
        return -1;
#  endif